SSE consolidation + apps improvements + eslint fixes

jonty007 opened the pull request:
Add subscription/streaming behavior around streamTXOs and streamMempoolCleanup, mempool cleanup (including DB-oriented details and docs), SSE consolidation, chat and explorer touchpoints, wallet/transaction UI components, TBC20/TBC721 and swap/chess contract/build alignment, node schema + multi-chain setup examples, lib typings/tests for Computer, and shared test rerun script improvements.

Details

- API / docs: New or expanded docs for streaming and mempool cleanup; Node index updates.
- Runtime: Node DB schema addition; chain-setup env/conf examples kept in sync.
- Apps: Chat subscriptions; explorer minor hook-in.
- Packages: ESLint/package.json changes; built artifacts updated for several packages.

jonty007 committed:
eslint fixes

jonty007 committed:
test:rerun dev ex improvements and fixes

jonty007 committed:
lint fixes

ltardivo committed:
Explicitly exports Contract from lib

jonty007 committed:
updated package and package lock

jonty007 committed:
fix lib bundling

jonty007 committed:
merge commit

jonty007 committed:
explorer, eslint and ft2

ClemensLey commented on packages/components/built/DecodeTransaction.d.ts:
```diff
@@ -0,0 +1,4 @@
+export declare function DecodedransactionComponent(): import("react/jsx-runtime").JSX.Element;
```
We should rename in all relevant files:
* `DecodedransactionComponent` -> `DecodeTransactionComponent`
* `Decodedransaction` -> `DecodeTransaction`

jonty007 committed:
improve tooltip

jonty007 committed:
merge commit

jonty007 committed:
charts UI UX fix

jonty007 committed:
build file updates

jonty007 committed:
merge commit

jonty007 committed:
merge commit

jonty007 committed:
Order book UI fixes, components enhancements

ClemensLey committed:
remove crypto-js dependency

ClemensLey committed:
update parcel and zeromq versions

ClemensLey committed:
Update development.md

ClemensLey committed:
Update dockerfile to avoid building from source

ClemensLey committed:
built files

jonty007 committed:
update deploy commands to support http2 node made UI/UX fix for exchange and live streaming

jonty007 committed:
build fixes

jonty007 committed:
env, doc and docker compose update

jonty007 committed:
merge of single event source

ClemensLey committed:
feat!(lib): introduce SmartContract<T> generic and full TypeScript overhaul

- Add SmartContract<T> wrapper + Lifted<T> types for strongly-typed smart contracts
- Make Computer.new(), sync(), mint(), transfer() etc. generic (e.g. sync<typeof Token>)
- computer.encode() now returns tx: Transaction | null (requires explicit checks)
- Remove all @ts-ignore comments and unsafe casts across the monorepo
- Update TBC20, TBC721, chess-contracts, dao-contracts, swap, nft, explorer, vite-template
- Extract complete type definitions to packages/lib/types/
- Refactor mocks (PaymentMock now extends Mock) and helpers
- Complete re-implementation of lib based on a clear theoretical model

BREAKING CHANGE: Public APIs now use generics and encode() can return null.
Existing code relying on type assertions or raw class types will need updating.

ClemensLey committed:
built files

jonty007 committed:
remove http2 configuration and withdraw to self

jonty007 committed:
vite template fix, new types for lib

ltardivo committed:
Fixes vitest.puppeteer.config.ts

ltardivo committed:
Exclude Contract from lib exports

ltardivo committed:
Adds entries to "common issues" section

ClemensLey committed:
fix type errors in chess app and contracts

ClemensLey committed:
dist files

ClemensLey committed:
feat(lib): introduce `precise<T>` type assertion for SmartContract

- Add `precise<T>(value: SmartContract<any>)` type assertion (narrows `SmartContract<any>` → `SmartContract<T>`)
- Export `precise` + `lifted` from main types index
- Improve `LiftedContract` definition with `LiftedConstructorParams` + constructor signatures
- Replace unsafe `as unknown as Class` casts with `SmartContract<typeof Class>` across:
  - ChessBoard.tsx / StartGame.tsx (chess-app)
  - ChessContractHelper (chess-contracts)
  - stream tests
  - nextjs-template & vite-template mint pages
- Use `precise<typeof WinnerTxWrapper>` for clean redeem flow typing
- Minor: simplify `bigint` wallet balance casts

Improves type safety when working with on-chain smart contracts.

ltardivo committed:
Fixes types in docs

ltardivo committed:
Removes getTXOs and latest from inner-computer. Makes dao-contracts WIP. Includes description for smart contract invalidation when rev not found.

jonty007 committed:
wallet withdraw dynamic

jonty007 committed:
Merge branch 'staging' of https://github.com/bitcoin-computer/monorepo into feat/eslint-fixes

jonty007 committed:
UI/UX fixes for the exchange

jonty007 committed:
add built files

jonty007 committed:
PR request to fix bigdata related sorting and other issues

ltardivo committed:
Fixes types on chess-contracts

jonty007 committed:
remove http2 config and certs

ltardivo committed:
Fixes types

ltardivo committed:
Upload build files and bundles

ltardivo committed:
Uploads package-lock.json

ltardivo committed:
Fixes nextJs lint errors

ltardivo committed:
Fixes types

ltardivo committed:
Uploads package-lock.json

ltardivo committed:
Upload bundles and build files

ltardivo committed:
Lint fixes

ltardivo committed:
Upload bundles

ltardivo committed:
Merge resolution

ltardivo committed:
Removes deploy command in chess-app

ltardivo committed:
Fixes network configuration in components

ltardivo committed:
Allows to set wallet with different module specifier type

ltardivo committed:
Fixes rerun script

ltardivo committed:
Camelcase DecodedransactionComponent and Decodetransaction

ltardivo committed:
Updates error message for rev not found

ltardivo committed:
Inline tests

ltardivo committed:
Fixes test in node project
